src/Entity/AdherentEmailSubscriptionHistory.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Table(name="adherent_email_subscription_history")
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\AdherentEmailSubscriptionHistoryRepository")
+ *
+ * @Algolia\Index(autoIndex=false)
+ */
+class AdherentEmailSubscriptionHistory
+{
+    /**
+     * @var int
+     *
+     * @ORM\Column(type="bigint")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Adherent")
+     * @ORM\JoinColumn(onDelete="CASCADE")
```

Comment:
We should keep all information we have at a given time so it should be set adherent to `null` when one adherent is deleted.

Reply:
And IMO it should be explicit so we should not rely on that cascade stuff. The goal is to avoid silly mistakes.